File: bit manipulation/2433_find_the_original_array_of_prefix_xor.js
/**
 * @param {number[]} pref
 * @return {number[]}
 */
var findArray = function(pref) {

    const ans = [... pref]

    for(let i=1; i<pref.length; i++){
        ans[i] = pref[i-1] ^ pref[i]
    }

    return ans

};

File: bit manipulation/2433_find_the_original_array_of_prefix_xor.go
func findArray(pref []int) []int {

    ans := make([]int, len(pref))
    copy(ans, pref)

    for i :=1; i<len(pref); i++ {

        ans[i] = pref[i-1] ^ pref[i]
    }

    return ans

}